### DISCUSSION TITLE
1.0.0

### DISCUSSION THREAD

**Discussion by OStrama** [OStrama/weishaupt_modbus#29] [Announcements]
The new version 1.0.0 is out!

<hr /><em>This discussion was created from the release <a href='https://github.com/OStrama/weishaupt_modbus/releases/tag/1.0.0'>1.0.0</a>.</em>

**Comment by exp1338** [comment]
No Update from RC available via HA settings?

**Reply by MadOne** [reply]
Sometimes it takes some time for hacs to recognize a new version. If you are inpatient you can select reinstall in hacs.

**Reply by exp1338** [reply]
Thanks. I will wait. :)

**Reply by maddinhome** [reply]
Thank you for the great work. Because the WEMPortal is not always accessible. This is now the first choice for me. I hope Homeassistant includes this integration in the official release. So that it will continue to be supported in the future. It would be nice if the various heat pumps from Weishaupt could be supported. It would also be nice to have instructions on how I can create the corresponding configuration file myself (kennfield file). I'm having difficulty finding the relevant information at Weishaupt. That's why I'm very grateful that someone created a file for WSB 8.

**Reply by OStrama** [reply]
Thank you! The next step will be adding it to HACS officially.
And you are lucky, a file for the WSB8 already exists in the repository: https://github.com/OStrama/weishaupt_modbus/blob/main/weishaupt_wsb8_kennfeld.json
Please install it in the following way:

1. check in your installation where the default kennfeld JSON file has been stored by the integation. Usually this should be the Home Assistant config directory.
2. Copy the file "weishaupt_wsb8_kennfeld.json" to the same directory
3. Re-configure the integration and choose the new filename in the configuration dialog.
4. Restart Home Assistant to be sure that the file is used

**Reply by OStrama** [reply]
here's a description how to generate the power mapping file:

To create a new file, you should take the one generated by the integration as template and modify the values only.
Additionally, you will need the documentation of your heatpump that you can download from the Weishaupt website. In the documentation you'll find two plots called "Heizleistung bei Wasseraustrittstemperatur 35°C" and "Heizleistung bei Wasseraustrittstemperatur 55°C" The values of the variables in the file are taken from these two graphs.

The file contains three variables:

1. "known_t": [35, 55] : The values correspond to the temperatures of the graphs. Up to now, we only saw two graphs, one for 35 degree one fro 55. More values are allowed in principle.
2. "known_x": [-30, -25, -22, -20, -15, -10, -5, 0, 5, 10, 15, 20, 25, 30, 35, 40] : These are the points on the X-axis of the graphs where you would need to read out the heating power values. Steps of 5 degree seem to be sufficient to interpolate a smooth curve. The more values you read out, the better fit the curve.
3. "known_y": [[5700, 5700, 5700, 5700, 6290, 7580, 8660, 9625, 10300, 10580, 10750, 10790, 10830, 11000, 11000, 11000], 
    [5700, 5700, 5700, 5700, 6860, 7300, 8150, 9500, 10300, 10580, 10750, 10790, 10830, 11000, 11000, 11000]],  : These are two vectors that represent the power values read out at the X-Points mentioned in 2.  The first line represents the values of the graph with the first temperature value in known_t, the second line corresponds with the 2nd entry of known_t.

Format:
It is crucial that the format of the file stays unchanged, so every bracket, comma exclamation mark etc has it's meaning. Deleting one of them or adding others will result in an error.
The number of values in known_t must be similar with the number of lines in known_y.
The number of values in known_x must be similar to the number of values of each line in known_y

If you generated a new file, please create a pull request or simply send us the file so that we can integrate it in the repository.

**Comment by maddinhome** [comment]
Thanks for the tip about the wsb8 file and where exactly it needs to be saved. I had already found the file and copied it wherever I thought it belonged. But now I know that this belongs in the Config folder. Then thanks for the instructions on how to create a new file. I no longer need it since there is already a file for me, but it will certainly be helpful for everyone else.

**Comment by Bertk** [comment]
Thank you for this great integration. There are some entity names which end with a '?' and also values have a dimension '?' or strings ending with ':'. I am not sure about the meaning.

<img width="474" alt="image" src="https://github.com/user-attachments/assets/b23d9b6e-517f-4b9b-ad89-b27ace25da4e">

<img width="286" alt="image" src="https://github.com/user-attachments/assets/c73fdd2a-d6e5-4a8a-a5ab-63cecaeb56a9">

<img width="292" alt="image" src="https://github.com/user-attachments/assets/d8fc4149-dcd7-4bf2-9c5b-af473d9548ff">

**Reply by MadOne** [reply]
The questionmark is because we are not sure yet. Not all modbus registers are documented yet

**Reply by OStrama** [reply]
.. and it is very likely, that a value of 32768 shows that this entry is not valid for the given configuration. So if it never changes you can ignore it..

**Comment by OStrama** [comment]
1.0.4 is out, so I close this discussion
